graylog2-web-interface/src/components/sidecars/configuration-forms/ConfigurationVariablesHelper.jsx

```diff
@@ -0,0 +1,151 @@
+            this.errorModal.open();
+          // Not in use, ask for confirmation
+          } else {
+            deleteConfirmationModal.open();
+          }
+        });
+      };
+
+      const _handleDeleteConfirm = () => {
+        ConfigurationVariableActions.delete(configVar)
+          .then(() => this._onSuccessfulUpdate());
+      };
+
+      const escapedName = `\${${configVar.name}}`;
+      variableRows.push(
+        <tr key={this._getId(configVar.id)}>
+          <td><code>{escapedName}</code></td>
+          <td>{configVar.description}</td>
+          <td style={{ textAlign: 'right' }}>
+            <BootstrapModalConfirm ref={(c) => { deleteConfirmationModal = c; }}
+                                   title="Delete Configuration Variable?"
+                                   onConfirm={_handleDeleteConfirm}
+                                   onCancel={() => {}}>
+              <p>Are you sure you want to remove the configuration variable <strong>{configVar.name}</strong>?</p>
+            </BootstrapModalConfirm>
+            <Button bsStyle="primary" bsSize="xsmall" onClick={_handleDeleteCheck}>
+              Delete
+            </Button>
+            <EditConfigurationVariableModal id={configVar.id}
+                                            name={configVar.name}
+                                            description={configVar.description}
+                                            content={configVar.content}
+                                            create={false}
+                                            saveConfigurationVariable={this._saveConfigurationVariable} />
+          </td>
+        </tr>,
+      );
+    });
+    return variableRows;
+  };
+
+  _isLoading = () => {
+    return !(this.state.configurationVariables);
+  };
+
+  _saveConfigurationVariable = (configurationVariable, callback) => {
+    ConfigurationVariableActions.save.triggerPromise(configurationVariable)
+      .then(() => this._onSuccessfulUpdate(callback));
+  };
+
+  _onSuccessfulUpdate = (callback) => {
+    if (typeof callback === 'function') {
+      callback();
+    }
+    this._reloadConfiguration();
+  };
+
+  render() {
+    if (this._isLoading()) {
+      return <Spinner />;
+    }
+
+    return (
+      <div>
+        <EditConfigurationVariableModal create
+                                        saveConfigurationVariable={this._saveConfigurationVariable} />
+        <br />
+        <br />
+        <br />
+        <div className={`table-responsive ${ConfigurationHelperStyle.tableMaxHeight}`}>
+          <Table responsive>
+            <thead>
+              <tr>
+                <th>Name</th>
+                <th>Description</th>
+                <th style={{ textAlign: 'right' }} >Actions</th>
+              </tr>
+            </thead>
+            <tbody>
+              {this._configurationVariableListFormatter()}
+              <BootstrapModalWrapper ref={(modal) => { this.errorModal = modal; }}>
```

Comment:
I was playing a bit with the modal and have a version that I like a bit better, what do you think?

Before:
<img width="710" alt="screenshot 2018-11-26 at 14 32 48" src="https://user-images.githubusercontent.com/716185/49017144-63fa6280-f188-11e8-8f25-d02e2aa6fee1.png">


After:
<img width="681" alt="screenshot 2018-11-26 at 14 32 18" src="https://user-images.githubusercontent.com/716185/49017153-68268000-f188-11e8-9a9c-8444ea291e8a.png">


```js
              <BootstrapModalWrapper ref={(modal) => { this.errorModal = modal; }}>
                <Modal.Header>
                  <Modal.Title>Error deleting configuration variable</Modal.Title>
                </Modal.Header>
                <Modal.Body>
                  <Alert bsStyle="warning">
                    <p>
                      Cannot delete this configuration variable as it is still in use. Please remove the variable from
                      the following configurations and try again.
                      {this.state.errorModalContent}
                    </p>
                  </Alert>
                </Modal.Body>
                <Modal.Footer>
                  <Button onClick={() => { this.errorModal.close(); }}>Close</Button>
                </Modal.Footer>
              </BootstrapModalWrapper>
```